ikarus/finiteelements/mechanics/enhancedassumedstrains.hh

```diff
@@ -184,44 +150,86 @@ forward the
    * \param numberOfEASParameters The number of EAS parameters
    */
   void setEASType(int numberOfEASParameters) {
-    const auto& numberOfNodes = DisplacementBasedElement::numberOfNodes();
-    if (not((numberOfNodes == 4 and Traits::mydim == 2) or (numberOfNodes == 8 and Traits::mydim == 3)) and
-        (not isDisplacementBased()))
-      DUNE_THROW(Dune::NotImplemented, "EAS only supported for Q1 or H1 elements");
-    easVariant_.setEASType(numberOfEASParameters, localView().element().geometry());
+    check();
+    easVariant_.setEASType(numberOfEASParameters, underlying().localView().element().geometry());
   }
 
 protected:
+  inline void check() const {
```

Comment:
I think the check is enough if its checked when setting `numberofEAS`

Reply:
How shoul,d this be done only once? I you call `calculate...` you never know, if you are in a valid state or not.

Reply:
> maybe `checkInput()` or can we even check for Q1 / H1 at compile time?
Grid export if they contain only one geometry type at compile time. This can then be checked  and if the grid does contain different types a fallback to a runtime check can be implemented. This would also be nice for the finite elements, where we then know that for a Q1 element some vectors are exaclty 4,8,12 of size at compile time.
https://gitlab.dune-project.org/core/dune-grid/-/blame/master/dune/grid/test/gridcheck.hh#L277